tests/python/pants_test/engine/examples/parsers.py

```diff
@@ -79,7 +80,7 @@ def parse(self, filepath, filecontent):
     This includes `namedtuple` subtypes as well as any custom class with an `_asdict` method defined;
     see :class:`pants.engine.serializable.Serializable`.
     """
-    json = filecontent
+    json = ensure_text(filecontent)
```

Comment:
I spent a lot of time on this part and am not 100% convinced I chose the best answer, although this works.

`filecontent` was always unicode, except for a couple times it would equal `b''`, which I could never find the source for. So `encode()` and `decode()` don't work, since it's mixed.

In general, I think we wanted `filecontent` to be binary type, but then with this being a JSON parser unicode makes sense and almost all our use cases had it as unicode.

Either way, I think this file is only used in testing, right, because it falls under test/examples?

Reply:
Yea, ensure_text should be fine. As you said, only used for testing.